Kernel/Process.h

```diff
@@ -248,11 +253,11 @@ class Process final
 
     // Breakable iteration functions
     template<IteratorFunction<Process&> Callback>
-    static void for_each(Callback);
-    template<IteratorFunction<Process&> Callback>
-    static void for_each_in_pgrp(ProcessGroupID, Callback);
-    template<IteratorFunction<Process&> Callback>
-    void for_each_child(Callback);
+    static void for_each_ignoring_jails(Callback);
+
+    ErrorOr<void> for_each_with_jail_respect(Function<ErrorOr<void>(Process&)>);
```

Comment:
should `for_each_with_jail_respect` be static?

Reply:
I rather use it with `Process::current()` prefix to indicate it relies on it.

Reply:
On second thought, we could just put it inside the method and it will almost look the same, so let's do that...